### ISSUE TITLE
how to disable company-mode in eshell?

### ISSUE THREAD

**Issue by dzhwinter** [syl20bnr/spacemacs#2390]
Of course, auto disable, not manual disable it after i open eshell.
i have try "  (spacemacs|disable-company eshell-mode)" into my spacemacs file as doc says. not work.
And I have checked manual  substitute  the macro in core-auto-completion.el , still not work for me.   

master branch up-to-date 0.103.2
emacs 24.5.1
ubuntu 14.04
Thank you

**Comment by zilongshanren** [comment]
+1 for this issue, it's very annoying.

I have tried to add another hook to eshell-mode and manually call `company-mode -1`.

But I failed again...

**Comment by cpaulik** [comment]
I think this is difficult right now. It seems that company mode is enabled in the post-config part of use-package. see [here](https://github.com/syl20bnr/spacemacs/blob/develop/contrib/shell/packages.el#L30). This means that it basically runs last. So any hook you add will be overridden. This might be a bug?

**Comment by person808** [comment]
In `dotspacemacs/config`: `(spacemacs|disable-company eshell-mode)`

**Comment by cpaulik** [comment]
@person808 That works, thanks. @dzhwinter does it still not work for you?

**Comment by dzhwinter** [comment]
@cpaulik.Thanks, but still not work for me. I tried his method when i first encountered this problem. I had check it a third time. In dotspacemacs/config: (spacemacs|disable-company eshell-mode) is useless. It is an instinct idea that disable company after load all the configuration. But We failed again and again, maybe the idle time is the stuff? see https://github.com/syl20bnr/spacemacs/blob/develop/contrib/shell/packages.el#L32
